openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -781,149 +741,143 @@ def create_library_block(library_key, block_type, definition_id):
-        delete block. This should always be true except when this function
-        calls itself recursively.
-    """
-    def_key, lib_bundle = _lookup_usage_key(usage_key)
-    # Create a draft:
-    draft_uuid = get_or_create_bundle_draft(def_key.bundle_uuid, DRAFT_NAME).uuid
-    # Does this block have a parent?
-    if usage_key not in lib_bundle.get_top_level_usages() and remove_from_parent:
-        # Yes: this is not a top-level block.
-        # First need to modify the parent to remove this block as a child.
-        raise NotImplementedError
-    # Does this block have children?
-    block = load_block(usage_key, user=None)
-    if block.has_children:
-        # Next, recursively call delete_library_block(...) on each child usage
-        for child_usage in block.children:
-            # Specify remove_from_parent=False to avoid unnecessary work to
-            # modify this block's children list when deleting each child, since
-            # we're going to delete this block anyways.
-            delete_library_block(child_usage, remove_from_parent=False)
-    # Delete the definition:
-    if def_key.bundle_uuid == lib_bundle.bundle_uuid:
-        # This definition is in the library, so delete it:
-        path_prefix = lib_bundle.olx_prefix(def_key)
-        for bundle_file in get_bundle_files(def_key.bundle_uuid, use_draft=DRAFT_NAME):
-            if bundle_file.path.startswith(path_prefix):
-                # Delete this file, within this definition's "folder"
-                write_draft_file(draft_uuid, bundle_file.path, contents=None)
-    else:
-        # The definition must be in a linked bundle, so we don't want to delete
-        # it; just the <xblock-include /> in the parent, which was already
-        # deleted above.
-        pass
-    # Clear the bundle cache so everyone sees the deleted block immediately:
-    lib_bundle.cache.clear()
-    LIBRARY_BLOCK_DELETED.send_event(
-        library_block=LibraryBlockData(
-            library_key=lib_bundle.library_key,
-            usage_key=usage_key
-        )
+    Does a Component exist for this usage key?
+
+    This is a lower-level function that will return True if a Component object
+    exists, even if it was soft-deleted, and there is no active draft version.
+    """
+    try:
+        get_component_from_usage_key(usage_key)
+    except ObjectDoesNotExist:
+        return False
+    return True
+
+
+def get_or_create_olx_media_type(block_type: str) -> MediaType:
+    """
+    Get or create a MediaType for the block type.
+
+    Learning Core stores all Content with a Media Type (a.k.a. MIME type). For
+    OLX, we use the "application/vnd.*" convention, per RFC 6838.
+    """
+    return contents_api.get_or_create_media_type(
+        f"application/vnd.openedx.xblock.v1.{block_type}+xml"
     )
 
 
-def create_library_block_child(parent_usage_key, block_type, definition_id) -> LibraryXBlockMetadata:
+def create_component_for_block(content_lib, usage_key):
```

Comment:
Is this being called in a transaction.atomic?

Reply:
@bradenmacdonald I don't remember what the discussion here was, do you?

Reply:
No, sorry.

Reply:
I'll a TODO. I doubt it needs to block this merge.

Reply:
Wrapped this in `with transaction.atomic()`